### ISSUE TITLE
java.lang.module.FindException: Module javafx.web not found

### ISSUE THREAD

**Issue by AspireONE-zz** [AdoptOpenJDK/IcedTea-Web#827]
Hello, I'm trying to run JOSM ( https://josm.openstreetmap.de/ ) under Linux Fedora 34 using IcedTea-web. But it does not work (I used to work with Fedora 24 before I upgraded) I have filled bug report with JOSM and they redirected me there. I do have openjfx installed. JOSM does do work using Oracle's JRE on this system.

% rpm -q openjfx
openjfx-11.0.9.2-5.fc34.x86_64
%file /usr/lib/jvm/openjfx/javafx.web.jar
/usr/lib/jvm/openjfx/javafx.web.jar: Java archive data (JAR)

-------------------------------------------------------------------------
$ /usr/libexec/icedtea-web/javaws  -verbose "https://josm.openstreetmap.de/download/josm-latest.jnlp"
itw-rust-debug: checking deployment.log.file in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.file in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.file in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.file in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.stdstreams in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.stdstreams in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.stdstreams in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.stdstreams in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.system in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.system in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.system in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.system in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.user.logdir in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.user.logdir in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.user.logdir in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.user.logdir in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
/usr/libexec/icedtea-web/javaws
itw-rust-debug: itw mode: DISTRIBUTION
itw-rust-debug: trying jdk over properties (deployment.jre.dir)
itw-rust-debug: checking deployment.jre.dir in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.jre.dir in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.jre.dir in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.jre.dir in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: nothing
itw-rust-debug: trying jdk JAVA_HOME
itw-rust-debug: nothing (likely correct)
itw-rust-debug: trying jdk from registry
itw-rust-debug: nothing
itw-rust-debug: trying jdk from path
itw-rust-debug: skipping jdk from path, your build is distribution
itw-rust-debug: nothing
itw-rust-debug: failing down to hardcoded
selected jre: /usr/lib/jvm/jre-11-openjdk
//usr/share/icedtea-web/javaws.jar
/usr/share/java/rhino.jar
/usr/share/java/tagsoup.jar
/usr/share/java/IPAddress/ipaddress.jar
itw-rust-debug: checking deployment.launcher.rust.bootcp.remove in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.remove in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.remove in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.remove in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.add in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.add in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.add in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.add in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.remove in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.remove in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.remove in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.remove in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.add in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.add in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.add in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.add in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: used boot classpath: -Xbootclasspath/a://usr/share/icedtea-web/javaws.jar:/usr/share/java/rhino.jar:/usr/share/java/tagsoup.jar:/usr/share/java/IPAddress/ipaddress.jar:/usr/lib/jvm/jre-11-openjdk/lib/ext/nashorn.jar:
itw-rust-debug: used classpath: /usr/lib/jvm/jre-11-openjdk/lib/rt.jar:/usr/lib/jvm/jre-11-openjdk/lib/ext/jfxrt.jar:
itw-rust-debug: current name: javaws
itw-rust-debug: current bin: /usr/libexec/icedtea-web/javaws
//usr/share/icedtea-web/javaws_splash.png
itw-rust-debug: alt-java found
itw-rust-debug: command "/usr/lib/jvm/jre-11-openjdk/bin/alt-java" "-version"
itw-rust-debug: detected jdk 9 or up
itw-rust-debug: modular jdk
//etc/java/icedtea-web/itw-modularjdk.args
itw-rust-debug: alt-java found
itw-rust-debug: command "/usr/lib/jvm/jre-11-openjdk/bin/alt-java" "-splash://usr/share/icedtea-web/javaws_splash.png" "@//etc/java/icedtea-web/itw-modularjdk.args" "-Xbootclasspath/a://usr/share/icedtea-web/javaws.jar:/usr/share/java/rhino.jar:/usr/share/java/tagsoup.jar:/usr/share/java/IPAddress/ipaddress.jar:/usr/lib/jvm/jre-11-openjdk/lib/ext/nashorn.jar:" "-classpath" "/usr/lib/jvm/jre-11-openjdk/lib/rt.jar:/usr/lib/jvm/jre-11-openjdk/lib/ext/jfxrt.jar:" "-Dicedtea-web.bin.name=javaws" "-Dicedtea-web.bin.location=/usr/libexec/icedtea-web/javaws" "net.sourceforge.jnlp.runtime.Boot" "-verbose" "https://josm.openstreetmap.de/download/josm-latest.jnlp"
WARNING: package javax.jnlp not in java.desktop
[ITW-CORE][2021-10-04 02:22:54.252 CEST][DEBUG][net.adoptopenjdk.icedteaweb.logging.LoggerFactory] init logger factory to net.sourceforge.jnlp.util.logging.OutputControllerLoggerFactory@5305068a
[ITW-CORE][2021-10-04 02:22:54.275 CEST][INFO ][net.sourceforge.jnlp.runtime.EnvironmentPrinter] OpenWebStartLauncher called with args: [-verbose, https://josm.openstreetmap.de/download/josm-latest.jnlp].
[ITW-CORE][2021-10-04 02:22:54.275 CEST][INFO ][net.sourceforge.jnlp.runtime.EnvironmentPrinter] OS: Linux
[ITW-CORE][2021-10-04 02:22:54.275 CEST][INFO ][net.sourceforge.jnlp.runtime.EnvironmentPrinter] Java Runtime Red Hat, Inc.-11.0.12
[ITW-CORE][2021-10-04 02:22:54.285 CEST][DEBUG][net.sourceforge.jnlp.runtime.JNLPRuntime] using DEFAULT extension point
[ITW-CORE][2021-10-04 02:22:55.550 CEST][INFO ][net.sourceforge.jnlp.config.DeploymentConfiguration] Ico provider registered correctly.
[ITW-CORE][2021-10-04 02:22:55.601 CEST][ERROR][net.sourceforge.jnlp.config.DeploymentConfiguration]
Failed to parse property file /home/adam/.config/icedtea-web/deployment.properties
java.io.FileNotFoundException: /home/adam/.config/icedtea-web/deployment.properties (Nie ma takiego pliku ani katalogu)
	at java.base/java.io.FileInputStream.open0(Native Method)
	at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
	at java.base/java.io.FileInputStream.<init>(FileInputStream.java:112)
	at java.base/sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:86)
	at java.base/sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:184)
	at net.adoptopenjdk.icedteaweb.http.CloseableConnection.getInputStream(CloseableConnection.java:58)
	at net.sourceforge.jnlp.config.DeploymentConfiguration.parsePropertiesFile(DeploymentConfiguration.java:579)
	at net.sourceforge.jnlp.config.DeploymentConfiguration.findSystemConfigFile(DeploymentConfiguration.java:393)
	at net.sourceforge.jnlp.config.DeploymentConfiguration.loadSystemProperties(DeploymentConfiguration.java:232)
	at net.sourceforge.jnlp.config.DeploymentConfiguration.load(DeploymentConfiguration.java:190)
	at net.sourceforge.jnlp.config.DeploymentConfiguration.load(DeploymentConfiguration.java:169)
	at net.sourceforge.jnlp.runtime.JNLPRuntime$DeploymentConfigurationHolder.<clinit>(JNLPRuntime.java:453)
	at net.sourceforge.jnlp.runtime.JNLPRuntime.getConfiguration(JNLPRuntime.java:481)
	at net.adoptopenjdk.icedteaweb.client.console.JavaConsole.canShowOnStartup(JavaConsole.java:205)
	at net.sourceforge.jnlp.runtime.JNLPRuntime.initialize(JNLPRuntime.java:230)
	at net.sourceforge.jnlp.runtime.Boot.runMain(Boot.java:258)
	at net.sourceforge.jnlp.runtime.Boot.mainWithReturnCode(Boot.java:132)
	at net.sourceforge.jnlp.runtime.Boot.main(Boot.java:114)

[ITW-CORE][2021-10-04 02:22:55.602 CEST][INFO ][net.sourceforge.jnlp.config.DeploymentConfiguration] checkUrl failed for file:/home/adam/.config/icedtea-web/deployment.properties reason: /home/adam/.config/icedtea-web/deployment.properties (Nie ma takiego pliku ani katalogu)
[ITW-CORE][2021-10-04 02:22:55.603 CEST][WARN ][net.sourceforge.jnlp.config.DeploymentConfiguration] No USER level deployment.properties found at file:/home/adam/.config/icedtea-web/deployment.properties.
[ITW-CORE][2021-10-04 02:22:55.603 CEST][WARN ][net.sourceforge.jnlp.config.DeploymentConfiguration] No USER level deployment.properties found at file:/home/adam/.config/icedtea-web/deployment.properties.
[ITW-CORE][2021-10-04 02:22:55.603 CEST][WARN ][net.sourceforge.jnlp.config.DeploymentConfiguration] Exception while loading comment form config file: /home/adam/.config/icedtea-web/deployment.properties (Nie ma takiego pliku ani katalogu)
[ITW-CORE][2021-10-04 02:22:55.603 CEST][WARN ][net.sourceforge.jnlp.config.DeploymentConfiguration] Exception while loading comment form config file: /home/adam/.config/icedtea-web/deployment.properties (Nie ma takiego pliku ani katalogu)
[ITW-CORE][2021-10-04 02:22:55.672 CEST][DEBUG][net.sourceforge.jnlp.util.logging.UnixSystemLog] System logger called with result of 0
[ITW-CORE][2021-10-04 02:22:55.696 CEST][DEBUG][net.sourceforge.jnlp.util.logging.UnixSystemLog] System logger called with result of 0
[ITW-CORE][2021-10-04 02:22:55.704 CEST][DEBUG][net.sourceforge.jnlp.util.logging.UnixSystemLog] System logger called with result of 0
[ITW-CORE][2021-10-04 02:22:55.724 CEST][DEBUG][net.sourceforge.jnlp.util.logging.UnixSystemLog] System logger called with result of 0
[ITW-CORE][2021-10-04 02:22:55.732 CEST][DEBUG][net.sourceforge.jnlp.util.logging.UnixSystemLog] System logger called with result of 0
[ITW-CORE][2021-10-04 02:22:55.739 CEST][DEBUG][net.sourceforge.jnlp.util.logging.UnixSystemLog] System logger called with result of 0
[ITW-CORE][2021-10-04 02:22:55.745 CEST][DEBUG][net.sourceforge.jnlp.util.logging.UnixSystemLog] System logger called with result of 0
[ITW-CORE][2021-10-04 02:22:55.750 CEST][DEBUG][net.sourceforge.jnlp.util.logging.UnixSystemLog] System logger called with result of 0
[ITW-CORE][2021-10-04 02:22:55.760 CEST][DEBUG][net.sourceforge.jnlp.util.logging.UnixSystemLog] System logger called with result of 0
[ITW-CORE][2021-10-04 02:22:55.859 CEST][DEBUG][net.adoptopenjdk.icedteaweb.client.parts.dialogs.security.SecurityDialogMessageHandler] Starting security dialog thread
[ITW-CORE][2021-10-04 02:22:55.964 CEST][DEBUG][net.sourceforge.jnlp.security.KeyStores] Keystore file /home/adam/.config/icedtea-web/security/trusted.clientcerts exists.
[ITW-CORE][2021-10-04 02:22:55.965 CEST][DEBUG][net.sourceforge.jnlp.security.SecurityUtil] Loading Keystore /home/adam/.config/icedtea-web/security/trusted.clientcerts
[ITW-CORE][2021-10-04 02:22:55.966 CEST][DEBUG][net.sourceforge.jnlp.security.KeystorePasswordAttempter] Operating Keystore /home/adam/.config/icedtea-web/security/trusted.clientcerts
[ITW-CORE][2021-10-04 02:22:55.969 CEST][DEBUG][net.sourceforge.jnlp.security.KeystorePasswordAttempter] Operating Keystore Unknown
[ITW-CORE][2021-10-04 02:22:55.971 CEST][DEBUG][net.sourceforge.jnlp.security.KeyStores] Keystore file /usr/lib/jvm/java-11-openjdk-11.0.12.0.7-4.fc34.x86_64/lib/security/trusted.jssecerts does not exists.
[ITW-CORE][2021-10-04 02:22:55.971 CEST][DEBUG][net.sourceforge.jnlp.security.SecurityUtil] Loading Keystore Unknown
[ITW-CORE][2021-10-04 02:22:55.972 CEST][DEBUG][net.sourceforge.jnlp.security.KeystorePasswordAttempter] Operating Keystore Unknown
[ITW-CORE][2021-10-04 02:22:55.972 CEST][DEBUG][net.sourceforge.jnlp.security.KeyStores] Keystore file /usr/lib/jvm/java-11-openjdk-11.0.12.0.7-4.fc34.x86_64/lib/security/trusted.certs does not exists.
[ITW-CORE][2021-10-04 02:22:55.972 CEST][DEBUG][net.sourceforge.jnlp.security.SecurityUtil] Loading Keystore Unknown
[ITW-CORE][2021-10-04 02:22:55.972 CEST][DEBUG][net.sourceforge.jnlp.security.KeystorePasswordAttempter] Operating Keystore Unknown
[ITW-CORE][2021-10-04 02:22:55.973 CEST][DEBUG][net.sourceforge.jnlp.security.KeyStores] Keystore file /home/adam/.config/icedtea-web/security/trusted.jssecerts exists.
[ITW-CORE][2021-10-04 02:22:55.974 CEST][DEBUG][net.sourceforge.jnlp.security.SecurityUtil] Loading Keystore /home/adam/.config/icedtea-web/security/trusted.jssecerts
[ITW-CORE][2021-10-04 02:22:55.974 CEST][DEBUG][net.sourceforge.jnlp.security.KeystorePasswordAttempter] Operating Keystore /home/adam/.config/icedtea-web/security/trusted.jssecerts
[ITW-CORE][2021-10-04 02:22:55.975 CEST][DEBUG][net.sourceforge.jnlp.security.KeyStores] Keystore file /home/adam/.config/icedtea-web/security/trusted.certs exists.
[ITW-CORE][2021-10-04 02:22:55.975 CEST][DEBUG][net.sourceforge.jnlp.security.SecurityUtil] Loading Keystore /home/adam/.config/icedtea-web/security/trusted.certs
[ITW-CORE][2021-10-04 02:22:55.975 CEST][DEBUG][net.sourceforge.jnlp.security.KeystorePasswordAttempter] Operating Keystore /home/adam/.config/icedtea-web/security/trusted.certs
[ITW-CORE][2021-10-04 02:22:55.980 CEST][DEBUG][net.sourceforge.jnlp.security.KeyStores] Keystore file /usr/lib/jvm/java-11-openjdk-11.0.12.0.7-4.fc34.x86_64/lib/security/jssecacerts does not exists.
[ITW-CORE][2021-10-04 02:22:55.980 CEST][DEBUG][net.sourceforge.jnlp.security.SecurityUtil] Loading Keystore Unknown
[ITW-CORE][2021-10-04 02:22:55.980 CEST][DEBUG][net.sourceforge.jnlp.security.KeystorePasswordAttempter] Operating Keystore Unknown
[ITW-CORE][2021-10-04 02:22:55.981 CEST][DEBUG][net.sourceforge.jnlp.security.KeyStores] Keystore file /usr/lib/jvm/java-11-openjdk-11.0.12.0.7-4.fc34.x86_64/lib/security/cacerts exists.
[ITW-CORE][2021-10-04 02:22:55.981 CEST][DEBUG][net.sourceforge.jnlp.security.SecurityUtil] Loading Keystore /usr/lib/jvm/java-11-openjdk-11.0.12.0.7-4.fc34.x86_64/lib/security/cacerts
[ITW-CORE][2021-10-04 02:22:55.982 CEST][DEBUG][net.sourceforge.jnlp.security.KeystorePasswordAttempter] Operating Keystore /usr/lib/jvm/java-11-openjdk-11.0.12.0.7-4.fc34.x86_64/lib/security/cacerts
[ITW-CORE][2021-10-04 02:22:56.091 CEST][DEBUG][net.sourceforge.jnlp.security.KeyStores] Keystore file /home/adam/.config/icedtea-web/security/trusted.jssecacerts exists.
[ITW-CORE][2021-10-04 02:22:56.091 CEST][DEBUG][net.sourceforge.jnlp.security.SecurityUtil] Loading Keystore /home/adam/.config/icedtea-web/security/trusted.jssecacerts
[ITW-CORE][2021-10-04 02:22:56.092 CEST][DEBUG][net.sourceforge.jnlp.security.KeystorePasswordAttempter] Operating Keystore /home/adam/.config/icedtea-web/security/trusted.jssecacerts
[ITW-CORE][2021-10-04 02:22:56.093 CEST][DEBUG][net.sourceforge.jnlp.security.KeyStores] Keystore file /home/adam/.config/icedtea-web/security/trusted.cacerts exists.
[ITW-CORE][2021-10-04 02:22:56.094 CEST][DEBUG][net.sourceforge.jnlp.security.SecurityUtil] Loading Keystore /home/adam/.config/icedtea-web/security/trusted.cacerts
[ITW-CORE][2021-10-04 02:22:56.095 CEST][DEBUG][net.sourceforge.jnlp.security.KeystorePasswordAttempter] Operating Keystore /home/adam/.config/icedtea-web/security/trusted.cacerts
[ITW-CORE][2021-10-04 02:22:56.097 CEST][DEBUG][net.sourceforge.jnlp.security.KeyStores] Keystore file /usr/lib/jvm/java-11-openjdk-11.0.12.0.7-4.fc34.x86_64/lib/security/trusted.clientcerts does not exists.
[ITW-CORE][2021-10-04 02:22:56.097 CEST][DEBUG][net.sourceforge.jnlp.security.SecurityUtil] Loading Keystore Unknown
[ITW-CORE][2021-10-04 02:22:56.098 CEST][DEBUG][net.sourceforge.jnlp.security.KeystorePasswordAttempter] Operating Keystore Unknown
[ITW-CORE][2021-10-04 02:22:56.099 CEST][DEBUG][net.sourceforge.jnlp.security.KeyStores] Keystore file /home/adam/.config/icedtea-web/security/trusted.clientcerts exists.
[ITW-CORE][2021-10-04 02:22:56.099 CEST][DEBUG][net.sourceforge.jnlp.security.SecurityUtil] Loading Keystore /home/adam/.config/icedtea-web/security/trusted.clientcerts
[ITW-CORE][2021-10-04 02:22:56.100 CEST][DEBUG][net.sourceforge.jnlp.security.KeystorePasswordAttempter] Operating Keystore /home/adam/.config/icedtea-web/security/trusted.clientcerts
[ITW-CORE][2021-10-04 02:22:56.113 CEST][INFO ][net.sourceforge.jnlp.proxy.browser.FirefoxPreferencesFinder] Using firefox's profiles file: /home/adam/.mozilla/firefox/profiles.ini
[ITW-CORE][2021-10-04 02:22:56.115 CEST][INFO ][net.sourceforge.jnlp.proxy.browser.FirefoxPreferencesFinder] Found preferences file: 
[ITW-CORE][2021-10-04 02:22:56.118 CEST][INFO ][net.sourceforge.jnlp.proxy.browser.FirefoxPreferencesParser] Read 259 entries from Firefox's preferences
[ITW-CORE][2021-10-04 02:22:56.120 CEST][INFO ][net.sourceforge.jnlp.runtime.Boot] Proceeding with jnlp
[ITW-CORE][2021-10-04 02:22:56.122 CEST][INFO ][net.sourceforge.jnlp.runtime.Boot] JNLP file location: https://josm.openstreetmap.de/download/josm-latest.jnlp
[ITW-CORE][2021-10-04 02:22:56.123 CEST][DEBUG][net.sourceforge.jnlp.services.ServiceUtil] call privileged method: getCodeBase
[ITW-CORE][2021-10-04 02:22:56.130 CEST][DEBUG][net.sourceforge.jnlp.services.ServiceUtil] result: null
[ITW-CORE][2021-10-04 02:22:56.133 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.ResourceTracker] Will add resource at location 'https://josm.openstreetmap.de/download/josm-latest.jnlp'
[ITW-CORE][2021-10-04 02:22:56.137 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.ResourceTracker] Will add resource 'josm-latest.jnlp'
[ITW-CORE][2021-10-04 02:22:56.142 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.ResourceHandler] Will check and maybe put into cache: josm-latest.jnlp
[ITW-CORE][2021-10-04 02:22:56.144 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.ResourceHandler] Download for josm-latest.jnlp has not been started until now
[ITW-CORE][2021-10-04 02:22:56.145 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.ResourceHandler] Checking download state of josm-latest.jnlp
[ITW-CORE][2021-10-04 02:22:56.146 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.ResourceHandler] Download has not been started yet: josm-latest.jnlp
[ITW-CORE][2021-10-04 02:22:56.153 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.ResourceHandler] Download of resource josm-latest.jnlp will start now!
[ITW-CORE][2021-10-04 02:22:56.171 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.cache.CacheEntry] isCached: remote size:2139 cached size:2139 -> true
[ITW-CORE][2021-10-04 02:22:56.172 CEST][INFO ][net.adoptopenjdk.icedteaweb.resources.cache.LeastRecentlyUsedCache] isCached: https://josm.openstreetmap.de/download/josm-latest.jnlp - (v: null) = true
[ITW-CORE][2021-10-04 02:22:56.173 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.initializer.UnversionedResourceInitializer] needsUpdateCheck: https://josm.openstreetmap.de/download/josm-latest.jnlp -> true
[ITW-CORE][2021-10-04 02:22:56.174 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.initializer.UnversionedResourceInitializer] Candidate URLs for location=https://josm.openstreetmap.de/download/josm-latest.jnlp version=null state=INCOMPLETE: [https://josm.openstreetmap.de/download/josm-latest.jnlp]
[ITW-CORE][2021-10-04 02:22:56.182 CEST][DEBUG][net.sourceforge.jnlp.proxy.JNLPProxySelector] Selecting proxy for: https://josm.openstreetmap.de/download/josm-latest.jnlp
[ITW-CORE][2021-10-04 02:22:56.183 CEST][DEBUG][net.sourceforge.jnlp.proxy.JNLPProxySelector] Selected proxies: [DIRECT]
[ITW-CORE][2021-10-04 02:22:56.205 CEST][DEBUG][net.sourceforge.jnlp.proxy.JNLPProxySelector] Selecting proxy for: socket://josm.openstreetmap.de:443
[ITW-CORE][2021-10-04 02:22:56.205 CEST][DEBUG][net.sourceforge.jnlp.proxy.JNLPProxySelector] Selected proxies: [DIRECT]
[ITW-CORE][2021-10-04 02:22:56.629 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.initializer.UrlProber] URL connection 'https://josm.openstreetmap.de/download/josm-latest.jnlp' header fields: {Keep-Alive=[timeout=5, max=100], Accept-Ranges=[bytes], null=[HTTP/1.1 200 OK], Strict-Transport-Security=[max-age=16070400; includeSubDomains], Server=[Apache/2.4.29 (Ubuntu)], ETag=["85b-5cd720ac31b3c"], Connection=[Keep-Alive], Last-Modified=[Sun, 03 Oct 2021 12:36:14 GMT], Content-Length=[2139], X-Clacks-Overhead=[GNU Terry Pratchett], Date=[Mon, 04 Oct 2021 00:22:56 GMT], Content-Type=[application/x-java-jnlp-file]}
[ITW-CORE][2021-10-04 02:22:56.630 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.initializer.BaseResourceInitializer] Best url for location=https://josm.openstreetmap.de/download/josm-latest.jnlp version=null state=INCOMPLETE is https://josm.openstreetmap.de/download/josm-latest.jnlp by HEAD
[ITW-CORE][2021-10-04 02:22:56.676 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.cache.CacheEntry] isCached: remote size:2139 cached size:2139 -> true
[ITW-CORE][2021-10-04 02:22:56.676 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.cache.CacheEntry] https://josm.openstreetmap.de/download/josm-latest.jnlp - version null: isCached true
[ITW-CORE][2021-10-04 02:22:56.677 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.cache.CacheEntry] https://josm.openstreetmap.de/download/josm-latest.jnlp - version null: lastModified cache:1633264574000 actual:1633264574000 -> true
[ITW-CORE][2021-10-04 02:22:56.678 CEST][INFO ][net.adoptopenjdk.icedteaweb.resources.cache.LeastRecentlyUsedCache] isUpToDate: https://josm.openstreetmap.de/download/josm-latest.jnlp - (v: null) = true
[ITW-CORE][2021-10-04 02:22:56.678 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.initializer.UnversionedResourceInitializer] needsUpdate: https://josm.openstreetmap.de/download/josm-latest.jnlp -> false
[ITW-CORE][2021-10-04 02:22:56.724 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.initializer.BaseResourceInitializer] Use cached version of resource location=https://josm.openstreetmap.de/download/josm-latest.jnlp version=null state=DOWNLOADED
[ITW-CORE][2021-10-04 02:22:56.725 CEST][DEBUG][net.adoptopenjdk.icedteaweb.resources.ResourceTracker] Download done. Shutting down executor
[ITW-CORE][2021-10-04 02:22:56.736 CEST][INFO ][net.adoptopenjdk.icedteaweb.xmlparser.MalformedXMLParser] Using MalformedXMLParser
[ITW-CORE][2021-10-04 02:22:56.982 CEST][INFO ][net.sourceforge.jnlp.Parser] Homepage: https://josm.openstreetmap.de
[ITW-CORE][2021-10-04 02:22:56.982 CEST][INFO ][net.sourceforge.jnlp.Parser] Description: Java OpenStreetMap editor
[ITW-CORE][2021-10-04 02:22:56.983 CEST][DEBUG][net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesReader] Jars not ready to provide attribute Application-Name
[ITW-CORE][2021-10-04 02:22:56.983 CEST][INFO ][net.sourceforge.jnlp.JNLPFile] Acceptable vendor tag found, contains: OpenStreetMap
[ITW-CORE][2021-10-04 02:22:56.988 CEST][INFO ][net.sourceforge.jnlp.Parser] good - your JRE - 11.0.12 - match requested JRE - 1.8+
[ITW-CORE][2021-10-04 02:22:56.994 CEST][DEBUG][net.sourceforge.jnlp.runtime.JNLPRuntime] Acquired shared lock on /tmp/adam/netx/locks/netx_running to indicate javaws is running
[ITW-CORE][2021-10-04 02:22:56.996 CEST][DEBUG][net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesReader] Jars not ready to provide attribute Application-Name
[ITW-CORE][2021-10-04 02:22:56.996 CEST][DEBUG][net.adoptopenjdk.icedteaweb.manifest.ManifestAttributesReader] Jars not ready to provide attribute Application-Name
[ITW-CORE][2021-10-04 02:22:57.004 CEST][DEBUG][net.sourceforge.jnlp.services.ServiceUtil] call privileged method: checkSingleInstanceRunning
[ITW-CORE][2021-10-04 02:22:57.004 CEST][DEBUG][net.sourceforge.jnlp.services.ServiceUtil] arg: net.sourceforge.jnlp.JNLPFile@183142fe
[ITW-CORE][2021-10-04 02:22:57.008 CEST][DEBUG][net.sourceforge.jnlp.services.ServiceUtil] result: null
[ITW-CORE][2021-10-04 02:22:57.011 CEST][INFO ][net.sourceforge.jnlp.ItwJvmLauncher] About to launch external with commands: '[/usr/libexec/icedtea-web/javaws, -J--add-modules=java.scripting,java.sql,javafx.controls,javafx.media,javafx.swing,javafx.web, -J--add-exports=java.base/sun.security.action=ALL-UNNAMED, -J--add-exports=java.desktop/com.apple.eawt=ALL-UNNAMED, -J--add-exports=java.desktop/com.sun.imageio.spi=ALL-UNNAMED, -J--add-exports=java.desktop/com.sun.imageio.plugins.jpeg=ALL-UNNAMED, -J--add-exports=javafx.graphics/com.sun.javafx.application=ALL-UNNAMED, -J--add-exports=jdk.deploy/com.sun.deploy.config=ALL-UNNAMED, -J--add-opens=java.base/java.lang=ALL-UNNAMED, -J--add-opens=java.base/java.nio=ALL-UNNAMED, -J--add-opens=java.base/jdk.internal.loader=ALL-UNNAMED, -J--add-opens=java.base/jdk.internal.ref=ALL-UNNAMED, -J--add-opens=java.desktop/javax.imageio.spi=ALL-UNNAMED, -J--add-opens=java.desktop/javax.swing.text.html=ALL-UNNAMED, -J--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED, -Xnofork, -verbose, https://josm.openstreetmap.de/download/josm-latest.jnlp]'
itw-rust-debug: checking deployment.log.file in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.file in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.file in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.file in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.stdstreams in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.stdstreams in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.stdstreams in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.stdstreams in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.system in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.system in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.system in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.log.system in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.user.logdir in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.user.logdir in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.user.logdir in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.user.logdir in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
/usr/libexec/icedtea-web/javaws
itw-rust-debug: itw mode: DISTRIBUTION
itw-rust-debug: trying jdk over properties (deployment.jre.dir)
itw-rust-debug: checking deployment.jre.dir in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.jre.dir in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.jre.dir in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.jre.dir in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: nothing
itw-rust-debug: trying jdk JAVA_HOME
itw-rust-debug: nothing (likely correct)
itw-rust-debug: trying jdk from registry
itw-rust-debug: nothing
itw-rust-debug: trying jdk from path
itw-rust-debug: skipping jdk from path, your build is distribution
itw-rust-debug: nothing
itw-rust-debug: failing down to hardcoded
selected jre: /usr/lib/jvm/jre-11-openjdk
//usr/share/icedtea-web/javaws.jar
/usr/share/java/rhino.jar
/usr/share/java/tagsoup.jar
/usr/share/java/IPAddress/ipaddress.jar
itw-rust-debug: checking deployment.launcher.rust.bootcp.remove in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.remove in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.remove in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.remove in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.add in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.add in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.add in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.bootcp.add in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.remove in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.remove in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.remove in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.remove in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.add in: /home/adam/.config/icedtea-web/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.add in: /home/adam/.icedtea/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.add in: /etc/.java/.deploy/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: checking deployment.launcher.rust.cp.add in: /etc/.java/deployment/deployment.properties
itw-rust-debug: property not located or file inaccessible
itw-rust-debug: used boot classpath: -Xbootclasspath/a://usr/share/icedtea-web/javaws.jar:/usr/share/java/rhino.jar:/usr/share/java/tagsoup.jar:/usr/share/java/IPAddress/ipaddress.jar:/usr/lib/jvm/jre-11-openjdk/lib/ext/nashorn.jar:
itw-rust-debug: used classpath: /usr/lib/jvm/jre-11-openjdk/lib/rt.jar:/usr/lib/jvm/jre-11-openjdk/lib/ext/jfxrt.jar:
itw-rust-debug: current name: javaws
itw-rust-debug: current bin: /usr/libexec/icedtea-web/javaws
//usr/share/icedtea-web/javaws_splash.png
itw-rust-debug: alt-java found
itw-rust-debug: command "/usr/lib/jvm/jre-11-openjdk/bin/alt-java" "-version"
itw-rust-debug: detected jdk 9 or up
itw-rust-debug: modular jdk
//etc/java/icedtea-web/itw-modularjdk.args
itw-rust-debug: alt-java found
itw-rust-debug: command "/usr/lib/jvm/jre-11-openjdk/bin/alt-java" "--add-modules=java.scripting,java.sql,javafx.controls,javafx.media,javafx.swing,javafx.web" "--add-exports=java.base/sun.security.action=ALL-UNNAMED" "--add-exports=java.desktop/com.apple.eawt=ALL-UNNAMED" "--add-exports=java.desktop/com.sun.imageio.spi=ALL-UNNAMED" "--add-exports=java.desktop/com.sun.imageio.plugins.jpeg=ALL-UNNAMED" "--add-exports=javafx.graphics/com.sun.javafx.application=ALL-UNNAMED" "--add-exports=jdk.deploy/com.sun.deploy.config=ALL-UNNAMED" "--add-opens=java.base/java.lang=ALL-UNNAMED" "--add-opens=java.base/java.nio=ALL-UNNAMED" "--add-opens=java.base/jdk.internal.loader=ALL-UNNAMED" "--add-opens=java.base/jdk.internal.ref=ALL-UNNAMED" "--add-opens=java.desktop/javax.imageio.spi=ALL-UNNAMED" "--add-opens=java.desktop/javax.swing.text.html=ALL-UNNAMED" "--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED" "-splash://usr/share/icedtea-web/javaws_splash.png" "@//etc/java/icedtea-web/itw-modularjdk.args" "-Xbootclasspath/a://usr/share/icedtea-web/javaws.jar:/usr/share/java/rhino.jar:/usr/share/java/tagsoup.jar:/usr/share/java/IPAddress/ipaddress.jar:/usr/lib/jvm/jre-11-openjdk/lib/ext/nashorn.jar:" "-classpath" "/usr/lib/jvm/jre-11-openjdk/lib/rt.jar:/usr/lib/jvm/jre-11-openjdk/lib/ext/jfxrt.jar:" "-Dicedtea-web.bin.name=javaws" "-Dicedtea-web.bin.location=/usr/libexec/icedtea-web/javaws" "net.sourceforge.jnlp.runtime.Boot" "-Xnofork" "-verbose" "https://josm.openstreetmap.de/download/josm-latest.jnlp"
Error occurred during initialization of boot layer
java.lang.module.FindException: Module javafx.web not found
[ITW-CORE][2021-10-04 02:22:57.288 CEST][DEBUG][net.sourceforge.jnlp.runtime.Boot] Exiting Boot.mainWithReturnCode() with 0
[ITW-CORE][2021-10-04 02:22:58.292 CEST][DEBUG][net.sourceforge.jnlp.runtime.JNLPRuntime] Release shared lock on /tmp/adam/netx/locks/netx_running
[ITW-CORE][2021-10-04 02:22:58.292 CEST][INFO ][net.adoptopenjdk.icedteaweb.resources.cache.LeastRecentlyUsedCache] No other instances of javaws are running

**Comment by JoergHW** [comment]
I came across the same issue. Azul is not part of the official Fedora repos. There is a potential workaround, which fails, but gets you in the position to derive the solution from the output of "javaws".
The workaround is to add "--module-path=/usr/lib/jvm/openjfx" (from openjfx package), but javaws puts this option AFTER all the "--add-modules=" on the resulting java command.
Derive solution: take output of "javaws --verbose" at the end where it says 'command "/usr/lib/jvm/jre-openjdk/bin/alt-java"' and rearrange the parameters to have "--module-path=..." BEFORE all the "--add-modules=" and it works without additional repos.
Q: what about putting that option in front or at least adding an option where to put the addition?
My command from which I derived the direct call to java: javaws --module-path=/usr/lib/jvm/openjfx -jnlp http://josm.openstreetmap.de/download/josm.jnlp

**Comment by AspireONE-zz** [comment]
Hello JoergHW,

I can confirm that your solution works on Fedora35. That is, I did :

"dnf install openjfx"

Then I ran

"javaws -verbose -jnlp http://josm.openstreetmap.de/download/josm.jnlp"

Then I inserted "--module-path=/usr/lib/jvm/openjfx" right after the
word alt-java taken from the output of the --verbose above. Then I
worked for me.

Your shorter version of the fix did not work for me though.

**Comment by sclassen** [comment]
@JoergHW
A quick question:
the `-module-path=/usr/lib/jvm/openjfx` you mention, do you pass it on the command line when calling javaws or is this specified in the JNLP file?

**Comment by JoergHW** [comment]
I put it on the commandline to javaws, but it gets pushed almost as the last resulting argument to the java call:
javaws --module-path=/usr/lib/jvm/openjfx -jnlp http://josm.openstreetmap.de/download/josm.jnlp

And since this does not work (wrong place for inserted argument), I need to copy & paste the output of "javaws -verbose ..." and move the "--module-path=..." to the front. (That's the actual fix, but I wanted to describe the way to get there. See previous post.)
I don't know if that's a bug in Java to extend the module-path only after it got mentioned, or if it was designed like this on purpose. (In good old days - and still now - there is no difference for "ls -lsrtc" and "ls -sctrl". All flags are parsed and then processing starts.)
